File: Java/app/src/main/java/visprolabweek1/Animal.java
package main.java.visprolabweek1;
// Konsep: Abstraction
// Menyembunyikan detail kompleks, hanya kasih lihat yang penting
abstract class Animal {
    protected String name;

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
    // Method abstract, tidak punya isi, harus di override di child class
    public abstract void makeSound();
}

File: Dart/dart_application_1/bin/animal.dart
// Konsep: Abstraction
// - abstract class: tidak bisa dibuat objek langsung.
// - Method abstract harus di-override oleh child.

abstract class Animal {
  String name = '';

  void setName(String name) {
    this.name = name;
  }

  // Method tanpa isi → harus diimplementasi oleh anak
  void makeSound();
}